hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2328,8 +2329,8 @@ func reconcileCAPIManagerRoleBinding(binding *rbacv1.RoleBinding, role *rbacv1.R
 	return nil
 }
 
-func reconcileCAPIProviderRole(role *rbacv1.Role) error {
-	role.Rules = []rbacv1.PolicyRule{
+func reconcileCAPIProviderRole(role *rbacv1.Role, p platform.Platform) error {
```

Comment:
this is disrupting existing platform/behaviour, can we please have a unit test for it?

Reply:
Added unit test, I used gomock to generate mock for the platform.Platform interface (gomock was also added to go.mod)

Reply:
Did you run something manually to generate platform_generated.go that now we need to automate via Makefile?
If so may be just keep it simple for now, drop gomock and use kubevirt platform for the unit test.

Reply:
This mock file is auto generated with `go generate ./...`
It can be added to the Makefile

Reply:
ok, so we'd need to have a target for it and include a dep for the unit tests and verify targets. I'm not sure this is adding much value at this point since the interface and the logic are fairly simple. I'd rather just rely on kubevirt platform to unit test which adds also implicit value and avoid adding a new dep.